p4-16/spec/P4-16-spec.mdk

```diff
@@ -4412,6 +4414,33 @@ select (p.tcp.port) {
 The expression `16w0 &&& 16w0xFC00` describes the set of 16-bit
 values whose most significant six bits are zero.
 
+Some targets may support parser value set, see Section [#sec-value-set]. Given
+a type `T` for the type parameter of the value set, the type of the value set
+is `set<T>`. The type of the value set must match to the type of all other
+`keysetExpression` in the same `select` expression. If there is a mismatch, the
+compiler must raise an error. The type of the values in the set must be one of
+bit<>, tuple, and struct.
+
+For example, to allow control plane API to specify TCP reserved ports at
+runtime, one could write:
+
+~ Begin P4Example
+struct vsk_t {
+    @match(ternary)
```

Comment:
Would it be a problem for the compiler, and/or the language definition, if this annotation name were 'match_kind' instead of 'match', so it was the same as 'match_kind' used elsewhere in P4_16?  This is a low priority issue, in my mind.  'match' is fine if there are any problems using 'match_kind' as an annotation name.

Reply:
match_kind is a keyword used to declare new match kinds. We may be able to tweak the grammar to make it work, but it may be confusing.

Reply:
Use match_kind triggers a frontend error. I will use match for now.
```
syntax error, unexpected MATCH_KIND
  @match_kind
   ^^^^^^^^^^
error: 1 errors encountered, aborting compilation```